### PR TITLE
Dev/backend model search and download

### PR DESCRIPTION
**Pull request by L-jasmine**
* Implemented search and download.

### CODE REVIEW

File: moxin-protocol/src/data.rs [hunk 1]
```diff
@@ -40,17 +44,17 @@ pub struct DownloadedFile {
...
+#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
 pub struct Model {
     pub id: ModelID,
     pub name: String,
     pub summary: String,
     pub size: String,
     pub requires: String,
     pub architecture: String,
-    pub released_at: NaiveDate,
+    pub released_at: DateTime<Utc>,
```
Comment (L-jasmine) [inline moxin-protocol/src/data.rs:55]: The time data obtained from Hugging Face is in UTC, and I think it should be changed to UTC here. Then, when using it, convert it to NativeDate or NativeDateTime for better compatibility.

Comment (L-jasmine) [reply moxin-protocol/src/data.rs:55]: Are you okay with the modifications made to the released_at?

@jmbejar

Comment (jmbejar) [reply moxin-protocol/src/data.rs:55]: Makes totally sense 👍

File: moxin-frontend/src/data/store.rs [hunk 2]
```diff
@@ -104,8 +104,8 @@ impl Store {
     }
 
     pub fn formatted_model_release_date(model: &Model) -> String {
-        let released_at = model.released_at.format("%b %-d, %C%y");
-        let days_ago = (Utc::now().date_naive() - model.released_at).num_days();
+        let released_at = model.released_at.naive_local().format("%b %-d, %C%y");
+        let days_ago = (Utc::now() - model.released_at).num_days();
```
[the remark this answers is not in the document]
Comment (jmbejar) [reply moxin-frontend/src/data/store.rs:166]: This is OK!

**Comment by jmbejar** [comment]
@L-jasmine I was testing as much as possible, so I'm ready to accept and merge the PR.

However, I noted the following issues or pending things (just want to double check if you're aware of them):

- Many bits of information pending in model records (size, require, architecture)
- Tags and other bit if information pending in file records

**Review by jmbejar — APPROVED** [review]

**Comment by L-jasmine** [comment]
> @L-jasmine I was testing as much as possible, so I'm ready to accept and merge the PR.
> 
> However, I noted the following issues or pending things (just want to double check if you're aware of them):
> 
> * Many bits of information pending in model records (size, require, architecture)
> * Tags and other bit if information pending in file records

These issues are in the process of being resolved.

**Pull request merged**